### ISSUE TITLE
req.body does not contain parsed data when JSON

### ISSUE THREAD

**Issue by james2doyle** [cyrilis/luvit-mooncake#3]
I was looking at the readme and it says that the [body gets parsed on post requests](https://github.com/cyrilis/luvit-mooncake#reqbody). So I tested this by sending some JSON in my post request. It would error out because my JSON data would go into the table key of `req.body`.

For example, if I sent `{hello:"World"}` then the `req.body` would spit back `req.body = { ['{hello:"World"}'] = {} }`. Looking at the source, I found that _all request bodies are parsed with querystring.parse_. Which is cool for form submissions, but it didn't seem in line with what your readme said.

I changed the following line https://github.com/cyrilis/luvit-mooncake/blob/master/index.lua#L173 to this:

``` lua
local bodyObj
-- is this request JSON?
if req.headers["Content-Type"] == 'application/json' then
  bodyObj = JSON.parse(body)
else
  -- ok treat it as a form
  bodyObj = querystring.parse(body)
end
```

I also added `local JSON = require("json")` at the head of the index file.

I can submit a proper PR for this, but I was wondering if there was maybe something I was missing?

**Comment by cyrilis** [comment]
Hi @james2doyle, 
Thanks for your feedback, but I didn't test it with JSON request, and I did forget to parse JSON data :grin: , that's my bad. 
And your changes look good, May you please submit a pull-request? 

Thanks.

**Comment by james2doyle** [comment]
Got it fixed in #4
